src/internal.c

```diff
@@ -8592,39 +8592,43 @@ int SendUserAuthRequest(WOLFSSH* ssh, byte authId, int addSig)
 
     if (ret == WS_SUCCESS) {
         if (ssh->ctx->userAuthCb != NULL) {
-            WLOG(WS_LOG_DEBUG, "SUAR: Calling the userauth callback");
+            int cbRet = WOLFSSH_USERAUTH_FAILURE;
 
+            WLOG(WS_LOG_DEBUG, "SUAR: Calling the userauth callback");
             WMEMSET(&authData, 0, sizeof(authData));
             authData.type = authId;
             authData.username = (const byte*)ssh->userName;
             authData.usernameSz = ssh->userNameSz;
 
             if (authId & WOLFSSH_USERAUTH_PASSWORD) {
-                ret = ssh->ctx->userAuthCb(WOLFSSH_USERAUTH_PASSWORD,
+                cbRet = ssh->ctx->userAuthCb(WOLFSSH_USERAUTH_PASSWORD,
                         &authData, ssh->userAuthCtx);
-                if (ret != WOLFSSH_USERAUTH_SUCCESS) {
+                printf("password cbRet = %d\n", cbRet);
```

Comment:
know this is a draft.... but had taken a peek at the changes, please do a grep for extra debug printfs before the final

Reply:
I will, eventually. That commit is tagged "WIP". My plan is to squash this all into one commit when I'm done. This is a PR to get the automated testing.